smart_open/s3.py

```diff
@@ -593,15 +597,16 @@ def iter_bucket(bucket_name, prefix='', accept_key=None,
     --------
 
       >>> # get all JSON files under "mybucket/foo/"
-      >>> for key, content in iter_bucket(bucket_name, prefix='foo/', accept_key=lambda key: key.endswith('.json')):
+      >>> for key, content in iter_bucket(bucket_name, prefix='foo/',
+      ...                                 accept_key=lambda key: key.endswith('.json')):
       ...     print key, len(content)
 
       >>> # limit to 10k files, using 32 parallel workers (default is 16)
       >>> for key, content in iter_bucket(bucket_name, key_limit=10000, workers=32):
       ...     print key, len(content)
     """
     if accept_key is None:
-        accept_key = lambda key: True
+        accept_key = _accept_all
```

Comment:
Why this obfuscation?

Reply:
PEP8 [discourages](https://pep8.org) from using lambda expressions other when they are unavoidable, like within function calls.

> Always use a def statement instead of an assignment statement that binds a lambda expression directly to an identifier.